capa/features/extractors/ghidra/helpers.py

```diff
@@ -70,3 +75,156 @@ def get_function_symbols() -> Iterator[ghidra.program.database.function.Function
     """yield all non-external function symbols"""
 
     yield from currentProgram.getFunctionManager().getFunctionsNoStubs(True)  # type: ignore [name-defined] # noqa: F821
+
+
+def get_file_imports() -> Dict[int, Any]:
+    """get all import names & addrs"""
+
+    addrs = []
+    names = []
+
+    for f in currentProgram.getFunctionManager().getExternalFunctions():  # type: ignore [name-defined] # noqa: F821
+        for r in f.getSymbol().getReferences():
+            if r.getReferenceType().isData():
+                addr = r.getFromAddress().getOffset()  # gets pointer to fake external addr
+
+        fstr = f.toString().split("::")  # format: MODULE.dll::import / MODULE::Ordinal_*
+        if "Ordinal_" in fstr[1]:
+            fstr[1] = f"#{fstr[1].split('_')[1]}"
+
+        for name in capa.features.extractors.helpers.generate_symbols(fstr[0][:-4], fstr[1]):
+            addrs.append(addr)
+            names.append(name)
+
+    return dict(zip(addrs, names))
+
+
+def get_file_externs() -> Dict[int, Any]:
+    addrs = []
+    names = []
+
+    for sym in currentProgram.getSymbolTable().getAllSymbols(True):  # type: ignore [name-defined] # noqa: F821
+        # .isExternal() misses more than this config for the function symbols
+        if sym.getSymbolType() == SymbolType.FUNCTION and sym.getSource() == SourceType.ANALYSIS and sym.isGlobal():
+            name = sym.getName()  # starts to resolve names based on Ghidra's FidDB
+            if name.startswith("FID_conflict:"):  # format: FID_conflict:<function-name>
+                name = name[13:]
+            addrs.append(sym.getAddress().getOffset())
+            names.append(name)
+            if name.startswith("_"):
+                # some linkers may prefix linked routines with a `_` to avoid name collisions.
+                # extract features for both the mangled and un-mangled representations.
+                # e.g. `_fwrite` -> `fwrite`
+                # see: https://stackoverflow.com/a/2628384/87207
+                names.append(name[1:])
+
+    return dict(zip(addrs, names))
+
+
+def map_fake_import_addrs() -> Dict[int, int]:
+    real_addrs = []
+    fake_addrs = []
+
+    for f in currentProgram.getFunctionManager().getExternalFunctions():  # type: ignore [name-defined] # noqa: F821
+        fake_addrs.append(f.getEntryPoint().getOffset())
+        for r in f.getSymbol().getReferences():
+            if r.getReferenceType().isData():
+                real_addrs.append(r.getFromAddress().getOffset())
+
+    return dict(zip(fake_addrs, real_addrs))
+
+
+def get_external_locs() -> List[int]:
```

Comment:
do you have an example of this? if so, let's add a comment pointing to the example for future reference.